develop/tutorials/articles/340-data-upgrades-and-verifiers/creating-an-upgrade-process-for-your-application.markdown

```diff
@@ -43,136 +154,180 @@ class modifies a column:
+which is in the [`com.liferay.portal.upgrade` module](https://repository.liferay.com/nexus/content/repositories/liferay-public-releases/com/liferay/com.liferay.portal.upgrade/)
+The interface declares a [`register` method](https://docs.liferay.com/portal/7.0/javadocs/modules/apps/foundation/portal/com.liferay.portal.upgrade/com/liferay/portal/upgrade/registry/UpgradeStepRegistrator.html)
+that the registrator must override. In that method, the registrator must
+implement all the module's upgrade registrations.
 
-**Note:** Be careful with upgrade steps with the same name in different packages. 
-This could result in the wrong version being used for the upgrade.
-(see `com.liferay.calendar.upgrade.v1_0_0.UpgradeCalendarBooking` 
-and `com.liferay.calendar.upgrade.v1_0_1.UpgradeCalendarBooking` as example).
+Each schema upgrade is represented by an upgrade registration. An upgrade
+registration is an abstraction for all the changes you need to apply to the
+database from one schema version to the next one.
 
-$$$
+Upgrade registrations are defined by the following values:
+
+- **Module's bundle symbolic name**
+- **Schema version to upgrade from** (as a `String`)
+- **Schema version to upgrade to** (as a `String`)
+- **List of upgrade steps**
+
+The registrator `DLWebUpgrade` registers two upgrades: one from version `0.0.0`
+to `0.0.1` and another from `0.0.1` to `1.0.0`.
 
-The upgrade process above results in the creation of three rows in the `Release`
-table, one per upgrade registration. In terms of data, each registration is
-represented by a row in the `Release_` table on portal’s database. You can check
-the database to verify the upgrades that were executed for each module.
+The `DLWebUpgrade` registrator's first registration is for upgrading the
+database from schema version `0.0.0` to `0.0.1`. Its list of upgrade steps
+contains one: `new DummyUpgradeStep()`. 
 
-Next, you can learn how to specify differing module versions and schema versions
-in your bundle.
+	registry.register(
+		"com.liferay.document.library.web", "0.0.0", "1.0.0",
+		new DummyUpgradeStep());
+
+The [`DummyUpgradeStep` class](https://github.com/liferay/liferay-portal/blob/7.0.1-ga2/portal-kernel/src/com/liferay/portal/kernel/upgrade/DummyUpgradeStep.java)
+provides an empty upgrade step. The `DLWebUpgrade` registrator defines this
+registration  so that the upgrade framework records the module's `1.0.0` schema
+version in @product@'s `Release_` table. 
+
+**Important**: Modules that use Service Builder *should not* define a
+registration for their initial database schema version, as Service Builder
+already records their schema versions to @product@'s `Release_` table. Modules
+that don't use Service Builder, however, *should* define a registration for
+their initial schema. 
+
+The `DLWebUpgrade` registrator's second registration (from schema version
```

Comment:
I don't know if this is the best example. The 0.0.1 step is something that only liferay provide. I think that it is better if we don't use a example with this step.